### ISSUE TITLE
`Illegal invocation` when using `encode`/`decode`

### ISSUE THREAD

**Issue by anonghuser** [browser-fs/core#1]
Currently there is an "Illegal invocation" error in browsers when calling the `utils.encode` function from `SyncStore.makeRootDirectory`.
In general, the way the function is exported, it does not have a properly bound `this` and so will fail. Not sure if/how it works in node and passes your tests.

[src/utils.ts](https://github.com/browser-fs/core/blob/66586ac/src/utils.ts#L247)

If you insist on a one-liner, perhaps something like this would work instead:
```
export const encode = globalThis.TextEncoder.prototype.encode.bind(new globalThis.TextEncoder());
```

And likewise for `decode`.

**Comment by dr-vortex** [comment]
Does c16320e fix it?

**Comment by dr-vortex** [comment]
Fixes published in [v0.0.12](https://github.com/browser-fs/core/releases/tag/v0.0.12)

**Comment by dr-vortex** [comment]
> Yes, the encode/decode exports are now good and working. Creating the root directory works fine too.
However there are places in the code that do not use the decode function, instead just calling .toString(), which used to work with Buffer but no longer does with Uint8Array. Technically a separate issue, but anyway, i'm writing here

Fixed in 2054a8e
